tst/CTA.WebForms2Blazor.Tests/Factories/ClassConverterFactoryTests.cs

```diff
@@ -0,0 +1,115 @@
+            _workspaceManager = new WorkspaceManagerService();
+            _workspaceManager.CreateSolutionFile();
+            _primaryProjectId = _workspaceManager.CreateProjectFile("TestProjectName", metadataReferences: _metadataReferences);
+        }
+
+        [TestCase(typeof(GlobalClassConverter), "Global.asax.cs",
+            @"namespace TestNamespace {
+                using System.Web;
+
+                public class Global : HttpApplication { }
+            }")]
+        [TestCase(typeof(HttpHandlerClassConverter), "TestHandler.cs",
+            @"namespace TestNamespace {
+                using System.Web;
+
+                public class TestHandler : IHttpHandler { }
+            }")]
+        [TestCase(typeof(HttpModuleClassConverter), "TestModule.cs",
+            @"namespace TestNamespace {
+                using System.Web;
+
+                public class TestModule : IHttpModule { }
+            }")]
+        [TestCase(typeof(PageCodeBehindClassConverter), "TestPage.aspx.cs",
+            @"namespace TestNamespace {
+                using System.Web.UI;
+
+                public class TestPage : Page { }
+            }")]
+        [TestCase(typeof(ControlCodeBehindClassConverter), "TestControl.ascx.cs",
+            @"namespace TestNamespace {
+                using System.Web.UI;
+
+                public class TestComponent : UserControl { }
+            }")]
+        [TestCase(typeof(MasterPageCodeBehindClassConverter), "TestMasterPage.Master.cs",
+            @"namespace TestNamespace {
+                using System.Web.UI;
+
+                public class TestMasterPage : MasterPage { }
+            }")]
+        [TestCase(typeof(UnknownClassConverter), "TestUnknownClass.cs",
+            @"namespace TestNamespace {
+                public class TestUnknownClass { }
+            }")]
+        public async Task Build_Recognizes_Class_Type(Type targetType, string testFileName, string testDocumentText)
+        {
+            var testDocumentPath = Path.Combine("C:", "Directory1", "Directory2", testFileName);
+            var did = _workspaceManager.AddDocument(_primaryProjectId, "TestDocument", testDocumentText);
+            var model = await _workspaceManager.GetCurrentDocumentSemanticModel(did);
+            var classConverter = _classConverterFactory.BuildMany(testDocumentPath, model).Single();
+
+            Assert.IsInstanceOf(targetType, classConverter);
+        }
+
+        [Test]
+        public async Task BuildMany_Splits_Classes()
+        {
+            var testDocumentPath = Path.Combine("C:", "Directory1", "Directory2", "TestDocumentName.cs");
+            var did = _workspaceManager.AddDocument(_primaryProjectId, "TestDocument", DocumentMultiClassText);
+            var model = await _workspaceManager.GetCurrentDocumentSemanticModel(did);
+            var classConverters = _classConverterFactory.BuildMany(testDocumentPath, model);
+
+            Assert.AreEqual(classConverters.Count(), 4);
```

Comment:
nit: Switch the parameter values in `AreEqual()`.

The signature of the assertion is `AreEqual(object expected, object actual)` which is important because if the assertion fails, it will print out:
```
Expected value: {expected}
Actual value: {actual}
```

So having the values switched could cause confusion.